### ISSUE TITLE
False positive using-constant-test for pandas.Index.is_unique

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#10400]
### Bug description

```python
`pylint` reports [`using-constant-test`](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/using-constant-test.html) for [`pandas.Index.is_unique`](https://pandas.pydata.org/docs/reference/api/pandas.Index.is_unique.html) (but not for [`pandas.Series.is_unique`](https://pandas.pydata.org/docs/reference/api/pandas.Series.is_unique.html)!)
=python
"z"
import pandas as pd

def i(l):
    "create index if values are unique"
    s = pd.Index(l)
    return s if s.is_unique else None
```

### Configuration

```ini

```

### Command used

```shell
pylint pylint.py
```

### Pylint output

```python
************* Module pylint
pylint.py:7:16: W0125: Using a conditional statement with a constant value (using-constant-test)
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.13.2 (tags/v3.13.2:4f8bb39, Feb  4 2025, 15:23:48) [MSC v.1942 64 bit (AMD64)]
```

### OS / Environment

windows

### Additional dependencies

```python
pandas==2.2.3
```

**Comment by sam-s** [comment]
no longer reproducible with
```
pylint 4.0.1
astroid 4.0.1
Python 3.14.0 (tags/v3.14.0:ebf955d, Oct  7 2025, 10:15:03) [MSC v.1944 64 bit (AMD64)]
```
thank you!

**Comment by Pierre-Sassoulas** [comment]
Thank you for the follow-up !
